### PR TITLE
feat: add workflow objects definitions, add sample unit workflow unit…

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Add definitions for workflow and related objects.

Keeping things to types/interface for now as it makes it a little easier to experiment with. We can consider moving to structured classes after a few iterations of resolving quirky kinks.

### Testing
Run `yarn workspace hmi-client run test`

### CODE REVIEW

File: packages/client/hmi-client/src/workflow/workflow.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+export interface WorkflowEdge {
+	id: string;
+	workflowId: string;
+	points: { x: number; y: number }[];
+
+	source: string;
+	sourcePort: number;
+
+	target: string;
+	targetPort: number;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/workflow.ts:56]:
```suggestion
	source: WorkflowNode["id"];
	sourcePort: number; // Are we really using integer like Petri Net?

	target: WorkflowNode["id"];
	targetPort: number;
```

File: packages/client/hmi-client/src/workflow/workflow.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	inputs: OperationData[];
+	outputs: OperationData[];
+}
+
+// Defines the data-exchange between WorkflowNode
+// In most cases the value here will be an assetId
+export interface WorkflowPort {
+	type: string;
+	value?: any;
+}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/workflow/workflow.ts:23]: an `id` would be useful, just to name them.
```js
inputs: [{ id: 'model-a', type: 'model' }, { id:'model-b', type: 'model' }],
```

**Review by echl — APPROVED** [review]

File: packages/client/hmi-client/src/workflow/workflow.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	y: number;
+	width: number;
+	height: number;
+
+	inputs: WorkflowPort[];
+	outputs: WorkflowPort[];
+
+	// FIXME: The section below is slated to be further spec'ed out later.
+	// State and progress, tracking of intermediate results
+	statusCode?: string; // not-run, completed, in-progress, error, invalid
```
[the remark this answers is not in the document]
Comment (echl) [reply packages/client/hmi-client/src/workflow/workflow.ts:43]: I hear daniel hates enums

File: packages/client/hmi-client/src/workflow/workflow.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	type: string;
+	value?: any;
+}
+
+// Node definition in the workflow
+// This is the graphical operation of the operation defined in operationType
+export interface WorkflowNode {
+	id: string;
+	workflowId: string;
+	operationType: string;
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/workflow/workflow.ts:30]: I don't think an enum works - enums are good for a fixed set but annoying to extend.

if you want it to be an extendable system an Enum does not work well. Imagine a 3rd party wants to write a model-optimization operation. We won't be able to accommodate that easily unless we already have the token as part of the enumeration already.

File: packages/client/hmi-client/src/workflow/workflow.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/workflow/workflow.ts:43]: premature engineering/optimization makes one lose focus on what is important? 😄

File: packages/client/hmi-client/src/workflow/workflow.ts (hunk 2 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/workflow/workflow.ts:23]: I will add it as an optional field now now, I thought about this before, but backed off ... I didn't think an id-scheme would be manageable given that the number of ports are dynamic at runtime, but will have to rethink it.

File: packages/client/hmi-client/src/workflow/workflow.ts (hunk 1 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/workflow/workflow.ts:56]: Nope. This is just direct numbering. Petrinet ascet's number system is indirect. e.g. to get value of index-2 in ascet is like `X[Y[2]]`

**Comment by YohannParis** [comment]
You could at least explain why you do not agree with the comments!

File: packages/client/hmi-client/src/workflow/workflow.ts (hunk 1 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/workflow/workflow.ts]: This is going to be so annoying

**Pull request merged**